### PR TITLE
add proxy_server & proxy_type to server install

### PR DESCRIPTION
**Pull request by sabo**
#### Pull Request (PR) description
Adds proxy parameters to the archive resource in install.pp, allowing prometheus to be installed from behind a proxy server.

#### This Pull Request (PR) fixes the following issues
PR #596 said it closed issue #186. However, while that PR adds `proxy_server` and `proxy_type` to Alertmanager and every exporter, it did not add the proxy parameters to the installation of prometheus itself.

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -49,6 +49,8 @@
   Optional[String[1]] $collect_tag                                              = $prometheus::collect_tag,
   Optional[Integer] $max_open_files                                             = $prometheus::max_open_files,
   Stdlib::Absolutepath $usershell                                               = $prometheus::usershell,
+  Optional[String[1]] $proxy_server                                             = $prometheus::proxy_server,
```
Comment (bastelfreak) [inline manifests/server.pp:52]: what kind of proxy address can be passed? an https URL? If so, can you please update the datatype for that? puppetlabs-stdlib has a few that might fit

**Comment by bastelfreak** [comment]
@sabo thanks for the PR! can you please add some unit tests for the change?

**Comment by schnitzel4** [comment]
great! we would like to use this feature

File: manifests/server.pp (hunk 1 above)
Comment (sabo) [reply manifests/server.pp:52]: this parameter and the below `$proxy_type` parameter are passed directly to the `archive` resource in install.pp, which accepts any string for `$proxy_server` and one of 'none', 'http', 'https', or 'ftp' for `$proxy_type`. So this matches those requirements (along with the datatypes of all the similar parameters on the manifests for the exporters)

**Comment by smortex** [comment]
> REFERENCE.md is outdated

Please update REFERENCE.md with

```sh-session
$ bundle exec rake strings:generate:reference
```

Add it to the PR and CI should continue.

**Pull request merged**